src/Manager.php

```diff
@@ -134,7 +134,8 @@ protected function gitBlame(Result $result, Settings $settings)
      */
     protected function getFilesFromPaths(array $paths, array $extensions, array $excluded = array())
     {
-        $extensions = array_flip($extensions);
+        $extensions = array_map('preg_quote', $extensions, array_fill(0, count($extensions), '`'));
```

Comment:
With `array_fill` you want create save extension list? Can you add test case for this code part, please?

Reply:
Just to be sure we are talking about the same thing, let me explain this code to you:

The `$extensions` is an array based on user input and may contain characters which have special meaning when used in a regular expression.

To still use this input in a regular expression, we need to escape any characters which have special meaning to make the input usable for the regular expression.
We use the PHP native [`preg_quote()`](https://www.php.net/manual/en/function.preg-quote.php) function to do this.
The `preg_quote()` function expects two parameters:
1. The string to escape;
2. The delimiter used for the regular expression (in this case: a backtick).

The more verbose way to do this would be:
```php
foreach ($extensions as $key => $extension) {
    $extensions[$key] = preg_quote($extension, '`');
}
```

Using [`array_map()`](https://www.php.net/manual/en/function.array-map) basically does the exact same thing as the above code snippet, but we need to pass it an array with the value to use for the second parameter for `preg_quote()`.

That's where the `array_fill(0, count($extensions), '`')` comes in. It just creates an array with the same amount of elements as `$extensions` with a backtick for the value of each, so the `preg_quote()` function gets both parameters passed.

The existing unit test already covers this, as the `php.dist` extension contains a `.`, which is a special character in regex and without the above escaping, it would match too match, i.e. it would also match `phptdist`, `php-dist` etc.
With the escaping in place, it will only match exactly `php.dist`.


Either way, I have added an additional unit test now to demonstrate and safeguard this.